apps/desktop/src/lib/stores/uniqueStore.ts

```diff
@@ -0,0 +1,75 @@
+import { writable, type Readable } from 'svelte/store';
+
+export function uniqueDerived<T>(source: Readable<T>) {
```

Comment:
I'm not sure the class is helpful, can't we just say:
```ts
export function shallowDeduplicate<A>(store: Readable<A>): Readable<A> {
	const deduplicatedStore = readable(get(store), (set) => {
		let lastValue = get(store);
		const unsubscribe = store.subscribe((value) => {
			if (!shallowCompare(lastValue, value)) {
				lastValue = value;
				set(value);
			}
		});
		return unsubscribe();
	});

	return deduplicatedStore;
}
```

Reply:
Let me try that out.

Reply:
You should be safe to set the readable initial value to undefined, and keep the result as `Readable<T>` because the start stop notifier gets run synchronously

Reply:
Hmm... or even write it as a derived to make that intent clearer

Reply:
```ts
export function shallowDeduplicate<A>(store: Readable<A>): Readable<A> {
	let lastValue: undefined | A = undefined;
	const deduplicatedStore = derived<Readable<A>, A>(store, (value, set) => {
			if (!shallowCompare(lastValue, value)) {
				lastValue = value;
				set(value);
			}
	});

	return deduplicatedStore;
}
```

Reply:
AHHhhhh no... that derived version isn't safe